BeagleBone_black/src/communications/communications.cpp

```diff
@@ -59,6 +59,9 @@ Communications::Communications(Logger& log, const char* ip, int portNo)
   }
 
   log_.INFO("COMN", "TCP/IP connection established.");
+  data::Communications cmn_data;
```

Comment:
Be careful about how this data structure is initialised, e.g. do you know what would the other values be? Either keep a master cope that you initialise first, or always read the shared copy before updating a single field.